examples/lie_labs.py

```diff
@@ -0,0 +1,87 @@
+import torch
+
+import theseus.labs.lie as lie
```

Comment:
@rmurai0610 thanks for the feedback!

Agreed we can include support for indexing. I wasn't clear on the second point about type consistency with the older Lie groups. Any thoughts @luisenp @fantaosha?

Reply:
Ah okay, makes sense to have consistent torch.Size, we can update that. I don't think it was intentional, @luisenp?

Reply:
Thanks for the feedback, @rmurai0610. This was not intentional, neither was the indexing. Will definitely fix this in the next batch of changes!

Reply:
Also, is there an easy way to create an identity group element with the new api?
Currently, to create an identity element of SO3, I can do 
`lie.cast(torch.eye(3)[None], lie.SO3)`
but it's quite verbose. Maybe we can have lie.SO3.identity() etc? (unless there is already a way to do this)

Also, this is personal preference, but in the older api, it was possible to write something like
```
g1 = th.SO3(tensor=...)
```
to make a new lie group element. 
With the new API, it's
```
g2 = lie.cast(..., lie.SO3) # or lie.from_tensor()
```
I feel these are not as intuitive, and isn't consitent with how other libraries like Sophus constructs a new lie group element.

Reply:
Yes we can identity function.

The cast/from_tensor are more torch like functions since this is meant to be a torch first library. We could also add sophus like functions and try to balance not making the api too messy. Any thoughts?

Reply:
Thank you, identity function would be quite useful (at least for some of my applications).

For the from_tensor, in pytorch, we can do
`x = torch.tensor([[1.0, 0.0, 0.0]])`
To me it feels more intuitive to follow torch's oop style, rather than having a functional style?

Also, the behavior of `torch.tensor` and `torch.from_numpy` is different (first one copies memory and latter does not)
```
import theseus.labs.lie as lie
import torch
import numpy as np

g1 = lie.SE3.rand(5)
g2 = lie.from_tensor(g1._t, lie.SE3)


print("g1 data_ptr", g1._t.data_ptr())
print("g2 data_ptr", g2._t.data_ptr())

t = np.array([0, 0, 0])
print("torch tensor 1", torch.tensor(t).data_ptr())
print("torch tensor 2", torch.tensor(t).data_ptr())
print("torch from_numpy 1", torch.from_numpy(t).data_ptr())
print("torch from_numpy 2", torch.from_numpy(t).data_ptr())
```

```
g1 data_ptr 5189291520
g2 data_ptr 5189291520 # points to same data as g2
torch tensor 1 5199293184
torch tensor 2 5199127232 # torch.tensor copied data elsewhere
torch from_numpy 1 105553175841472 
torch from_numpy 2 105553175841472 # torch.from_numpy points to the same data
```

Maybe the same convention should be followed for the library? (the lie.SE3 copies, whereas lie.cast/from_tensor does not)

Reply:
We are exposing both oop and functional style api. In that spirit, we could move such functional like methods to the `lieF` namespace and add missing methods on the oop `lie` namespace side.

Reply:
It’s my personal preference but I think something like
`lie.SE3(tensor=…)` and `lie.SE3.from_tensor(…)` in the oop version and `lieF.from_tensor` in the functional would be nice.
Being able to choose between copy and no-copy initialisation I think would be handy.

Reply:
I just realized that I didn't include in the example script something like `lie.LieTensor(t, lie.SO3)`, which was also allowed. The difference with `lie.from_tensor()` is that directly calling `LieTensor` returns a leaf tensor, but `from_tensor()` keeps the compute graph's history (see the image). 

<img width="562" alt="image" src="https://user-images.githubusercontent.com/4759586/230219090-865c82b7-e31b-4037-a7b5-82526f8ba7f4.png">

I like the suggestion of having `lie.SO3(tensor)` as well. I'll add this. 

Anything else I should consider?

Reply:
Regarding the copy vs. no-copy, at the moment it seems we don't have a way to create a lie tensor without making a copy. My understanding was that the implementation of `LieTensor` was such that the same underlying storage was used, but I'm running some tests and it doesn't seem to be the case (but I might be doing something wrong). I need to look more into this.